R/plotLoadings.R

```diff
@@ -0,0 +1,416 @@
+        } else {
+            # Ordering loadings and adding factor to keep the order
+            L <- .get_loadings_plot_data(loadings_matrix, n, ncomponents)
+            # Plot features with the layout selected
+            p <- .plot_pca_feature_loadings(L, layout, n, ncomponents)
+        }
+    return(p)
+    }
+)
+#' @rdname plotLoadings
+#' @export 
+setMethod("plotLoadings", signature = c(x = "matrix"),
+    function(x,
+            dimred = "PCA",
+            layout = "heatmap",
+            n = 10,
+            ncomponents = 5,
+            ...) {
+        # Making sure there is no error in parameters given by the user
+        .check_parameters(x,
+                        dimred = dimred,
+                        layout = layout,
+                        n = n,
+                        ncomponents = ncomponents,
+                        
+                        ...)
+                      
+        # Checking if there are enough components in the matrix
+        .check_components(x, ncomponents)
+        # Ordering loadings and adding factor to keep the order
+        df <- .get_loadings_plot_data(x, n, ncomponents)
+        # Plot features with the layout selected
+        p <- .plot_pca_feature_loadings(df, layout, n, ncomponents)
+        
+        return(p)
+    }
+)
+
+#' @rdname plotLoadings
+#' @export 
+setMethod("plotLoadings", signature = c(x = "SingleCellExperiment"),
+    function(x,
+            dimred = "PCA",
+            layout = "heatmap",
+            n = 10,
+            ncomponents = 5,
+            ...) {
+      
+        # Making sure there is no error in parameters given by the user
+        .check_parameters(x,
+                          dimred = dimred,
+                          layout = layout,
+                          n = n,
+                          ncomponents = ncomponents,
+                          ...)  
+        
+        loading_names <- c("rotation", "loadings")
+        attr_names <- names(attributes(reducedDim(x, dimred)))
+        attr_name <- attr_names[ attr_names %in% loading_names ]
+        if( length(attr_name) != 1 ) {
+            stop("Loadings cannot be found..")
+        }
+        loadings_matrix <- attr(reducedDim(x, dimred), attr_name)
+        # Checking if there are enough components in the matrix
+        .check_components(loadings_matrix, ncomponents)
+            
+        # Ordering loadings and adding factor to keep the order
+        L <- .get_loadings_plot_data(loadings_matrix, n, ncomponents)
+        # Plot features with the layout selected
+        p <- .plot_pca_feature_loadings(L, layout, n, ncomponents)
+        
+        return(p)
+    }
+)
+
+.check_parameters <- function(x, dimred, layout, n, ncomponents, tree.name, class, ...) {
+    # Check tree.name
+    if( is(x, "TreeSummarizedExperiment") && !(tree.name %in% rowTreeNames(x)  && .is_a_string(tree.name))){
+        stop("'tree.name' must be a single character value specifying a colTree.", call. = FALSE)
```

Comment:
rowTree

Reply:
Not all TreeSE have tree

Reply:
> Also this is failing while it should not
> 
> ```
> data("GlobalPatterns", package = "mia")
> tse <- GlobalPatterns
> tse <- transformAssay(tse, method = "clr", pseudocount = 1)
> tse <- agglomerateByPrevalence(tse, rank="Phylum", update.tree = TRUE)
> tse <- runPCA(tse, ncomponents = 5, assay.type = "clr")
> plotLoadings(tse, layout = "barplot")
> ```

I don't see why this would not work? It runs fine for me